src/atomvm_uf2create_provider.erl

```diff
@@ -139,62 +136,6 @@ parse_addr("16#" ++ AddrHex) ->
 parse_addr(AddrDec) ->
     list_to_integer(AddrDec).
 
-%%% UF2 defines
--define(UF2_MAGIC_START0, 16#0A324655).
--define(UF2_MAGIC_START1, 16#9E5D5157).
--define(UF2_MAGIC_END, 16#0AB16F30).
-
--define(UF2_FLAG_FAMILY_ID_PRESENT, 16#00002000).
-
-%%% Pico defines
--define(UF2_PICO_FLAGS, ?UF2_FLAG_FAMILY_ID_PRESENT).
--define(UF2_PICO_PAGE_SIZE, 256).
--define(UF2_PICO_FAMILY_ID, 16#E48BFF56).
-
-%%%
-
-do_uf2create(OutputPath, StartAddr, ImagePath) ->
-    {ok, ImageBin} = file:read_file(ImagePath),
-    BlocksCount0 = byte_size(ImageBin) div ?UF2_PICO_PAGE_SIZE,
-    BlocksCount =
-        BlocksCount0 +
-            if
-                byte_size(ImageBin) rem ?UF2_PICO_PAGE_SIZE =:= 0 -> 0;
-                true -> 1
-            end,
-    OutputBin = uf2create0(0, BlocksCount, StartAddr, ImageBin, []),
-    ok = file:write_file(OutputPath, OutputBin).
-
 %% @private
-uf2create0(_BlockIndex, _BlocksCount, _BaseAddr, <<>>, Acc) ->
-    lists:reverse(Acc);
-uf2create0(BlockIndex, BlocksCount, BaseAddr, ImageBin, Acc) ->
-    {PageBin, Tail} =
-        if
-            byte_size(ImageBin) >= ?UF2_PICO_PAGE_SIZE ->
-                split_binary(ImageBin, ?UF2_PICO_PAGE_SIZE);
-            true ->
-                {ImageBin, <<>>}
-        end,
-    PaddedData = pad_binary(PageBin, 476),
-    Block = [
-        <<
-            ?UF2_MAGIC_START0:32/little,
-            ?UF2_MAGIC_START1:32/little,
-            ?UF2_PICO_FLAGS:32/little,
-            BaseAddr:32/little,
-            ?UF2_PICO_PAGE_SIZE:32/little,
-            BlockIndex:32/little,
-            BlocksCount:32/little,
-            ?UF2_PICO_FAMILY_ID:32/little
-        >>,
-        PaddedData,
-        <<?UF2_MAGIC_END:32/little>>
-    ],
-    uf2create0(BlockIndex + 1, BlocksCount, BaseAddr + ?UF2_PICO_PAGE_SIZE, Tail, [Block | Acc]).
-
-%% @private
-pad_binary(Bin, Len) ->
-    PadCount = Len - byte_size(Bin),
-    Pad = binary:copy(<<0>>, PadCount),
-    [Bin, Pad].
+do_uf2create(OutputPath, StartAddr, ImagePath) ->
```

Comment:
It seems this is only called line 76 so I guess it could be inlined there.

Reply:
Absolutely. It also occurred to me there is really no reason to be making external system calls to check the os_type and mount point checks, there are already Erlang functions for these that would be Windows compatible too, if we ever end up with a Windows port. I think that was a rookie mistake on my part, mainly due to my unfamiliarity with OTP, since AtomVM was my introduction to the BEAM.

Reply:
I should have checked your exports! I just realized the `create` and `join` functions are not actually exported, which is fine. I just need to update this to use the exported `main/1` from uf2tool.

Reply:
Let me know if uf2tool needs to be updated to provide exports.

Reply:
Thanks, this is just as easy to use from the Erlang side. If I run into problems while continue working on an ExAtomVM task I will let you know.